roles/download/tasks/prep_download.yml

```diff
@@ -16,7 +16,7 @@
 - name: prep_download | Set image pull/info command for containerd
   set_fact:
     image_info_command: "{{ bin_dir }}/nerdctl -n k8s.io images --format '{% raw %}{{ '{{' }} .Repository {{ '}}' }}:{{ '{{' }} .Tag {{ '}}' }}{% endraw %}' 2>/dev/null | grep -v ^:$ | tr '\n' ','"
-    image_pull_command: "{{ bin_dir }}/nerdctl -n k8s.io pull --quiet"
+    image_pull_command: "{%- if containerd_insecure_registries is defined and containerd_insecure_registries != None -%}{{ bin_dir }}/nerdctl -n k8s.io pull --quiet --insecure-registry{%- else -%}{{ bin_dir }}/nerdctl -n k8s.io pull --quiet{%- endif -%}"
```

Comment:
how about just a simple `set_fact` before this that sets `nerdctl_extra_flags` to `---insecure-registry` and then add just `{{ nerdctl_extra_flags}}` at the end of the line ?

Reply:
i fixed code for your comments 

plz check thank

Reply:
Thanks for the quick response @mircyb!

/ok-to-test
/lgtm